plugins/parsers/xpath/parser.go

```diff
@@ -81,6 +68,14 @@ func (p *Parser) Init() error {
 		return fmt.Errorf("unknown data-format %q for xpath parser", p.Format)
 	}
 
+	// Make sure we do have a metric name
+	if p.DefaultMetricName == "" {
```

Comment:
Is the default metric name ever intended to be blank? Wouldn't this only happen if the plugin isn't registered correctly? If so, maybe Init() should fail here instead of trying to recover.

Reply:
The `DefaultMetricName` is injected by the config and this might be empty. Do you still want me to change it to an error?

Reply:
DefaultMetricName has `toml:"-"` so it doesn't look like it gets set by config. I thought it was only set when telegraf code registers the plugin. If that's the case I would prefer that we fail early and fix the registration, and not try to recover here.

Reply:
The value comes from the TOML config item `metric_name` and is used in the new style as argument to the `Parser.Creator` (defined with `Parser.Add` in `init`)

Reply:
Additionally it is still set though `InitFromConfig()` when deriving the parser the "old" way. But I'll remove the check and error out then.